core/sail/fts/lucene-api/src/main/java/org/eclipse/rdf4j/sail/lucene/LuceneSail.java

```diff
@@ -252,6 +257,33 @@
 	public static final String INCOMPLETE_QUERY_FAIL_KEY = "incompletequeryfail";
 
 	/**
+	 * Set this key to "false" to disable late evaluation of search queries.
+	 */
+	public static final String EVALUATION_MODE_KEY = "evaluationMode";
+
+	/**
+	 * Eagerly evaluates any search queries and then delegates to the base SAIL.
+	 */
+	public static final String EAGER_EVALUATION_MODE = "eager";
+
+	/**
+	 * Uses the base SAIL along with an embedded SERVICE to perform query evaluation. The SERVICE is used to
+	 * evaluate extended query algebra nodes such as {@link TupleFunction}s.
+	 */
+	public static final String SERVICE_EVALUATION_MODE = "service";
+
+	/**
+	 * Assumes the base SAIL supports an extended query algebra (e.g. {@link TupleFunction}s) and use it to
+	 * perform all query evaluation.
+	 */
+	public static final String NATIVE_EVALUATION_MODE = "native";
+
+	/**
+	 * Treats the base SAIL as a simple triple source and all the query evaluation is performed by this SAIL.
+	 */
+	public static final String TRIPLE_SOURCE_EVALUATION_MODE = "tripleSource";
```

Comment:
I'm not really sure I understand how 'native' and 'tripleSource' evaluation modes compare to 'eager'.